lib/source/s3.js

```diff
@@ -0,0 +1,284 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const Aws = require('aws-sdk');
+
+const DEFAULT_INTERVAL = 60000;
+
+class S3Parser {
+  constructor() {
+    this.properties = {};
+  }
+
+  update(data) {
+    let properties = {};
+
+    try {
+      properties = JSON.parse(data.toString());
+    } catch (e) {
+      // We should have an error condition but tbd
```

Comment:
I'm of two minds here. On the one hand, having an error condition and then testing for it is effectively testing the body that AWS sends back. 

On the other hand, it's not a terrible idea to test in isolation, and now that we can effectively stub `getObject` it can't hurt to do something here. 

I'm amenable either way.